specification/network/resource-manager/Microsoft.Network/stable/2019-09-01/networkSecurityGroup.json

```diff
@@ -801,6 +803,7 @@
           "description": "A collection of security rules of the network security group."
         },
         "defaultSecurityRules": {
+          "readOnly": true,
```

Comment:
this should be settable at creation time.

Reply:
I tried sending PUT request with custom rule inside `defaultSecurityRules` but it was overwritten. makes sense since it's read-only on backend

```
PS C:\Users\v-anevse> ARMClient.exe PUT /subscriptions/947d47b4-7883-4bb9-9d85-c5e8e2f572ce/resourceGroups/v-anevse/providers/Microsoft.Network/networkSecurityGroups/test-nsgro?api-version=2019-08-01 "{\`"location\`":\`"eastus\`",\`"properties\`":{\`"defaultSecurityRules\`":[{\`"name\`":\`"SomeTestRule\`",\`"properties\`":{\`"protocol\`":\`"*\`",\`"sourceAddressPrefix\`":\`"*\`",\`"destinationAddressPrefix\`":\`"*\`",\`"access\`":\`"Allow\`",\`"destinationPortRange\`":\`"80\`",\`"sourcePortRange\`":\`"*\`",\`"priority\`":130,\`"direction\`":\`"Inbound\`"}}]}}"

{
  "name": "test-nsgro",
  "id": "/subscriptions/947d47b4-7883-4bb9-9d85-c5e8e2f572ce/resourceGroups/v-anevse/providers/Microsoft.Network/networkSecurityGroups/test-nsgro",
  "etag": "W/\"f69a2d77-5d33-423a-abf0-fa9f925602ca\"",
  "type": "Microsoft.Network/networkSecurityGroups",
  "location": "eastus",
  "properties": {
    "provisioningState": "Updating",
    "resourceGuid": "a2c98794-04b0-4cee-8fde-90c5d77e70fe",
    "securityRules": [],
    "defaultSecurityRules": [
      {
        "name": "AllowVnetInBound",
        "id": "/subscriptions/947d47b4-7883-4bb9-9d85-c5e8e2f572ce/resourceGroups/v-anevse/providers/Microsoft.Network/networkSecurityGroups/test-nsgro/defaultSecurityRules/AllowVnetInBound",
        "etag": "W/\"f69a2d77-5d33-423a-abf0-fa9f925602ca\"",
        "type": "Microsoft.Network/networkSecurityGroups/defaultSecurityRules",
        "properties": {
          "provisioningState": "Updating",
          "description": "Allow inbound traffic from all VMs in VNET",
          "protocol": "*",
          "sourcePortRange": "*",
          "destinationPortRange": "*",
          "sourceAddressPrefix": "VirtualNetwork",
          "destinationAddressPrefix": "VirtualNetwork",
          "access": "Allow",
          "priority": 65000,
          "direction": "Inbound",
          "sourcePortRanges": [],
          "destinationPortRanges": [],
          "sourceAddressPrefixes": [],
          "destinationAddressPrefixes": []
        }
      },
      {
        "name": "AllowAzureLoadBalancerInBound",
        "id": "/subscriptions/947d47b4-7883-4bb9-9d85-c5e8e2f572ce/resourceGroups/v-anevse/providers/Microsoft.Network/networkSecurityGroups/test-nsgro/defaultSecurityRules/AllowAzureLoadBalancerInBound",
        "etag": "W/\"f69a2d77-5d33-423a-abf0-fa9f925602ca\"",
        "type": "Microsoft.Network/networkSecurityGroups/defaultSecurityRules",
        "properties": {
          "provisioningState": "Updating",
          "description": "Allow inbound traffic from azure load balancer",
          "protocol": "*",
          "sourcePortRange": "*",
          "destinationPortRange": "*",
          "sourceAddressPrefix": "AzureLoadBalancer",
          "destinationAddressPrefix": "*",
          "access": "Allow",
          "priority": 65001,
          "direction": "Inbound",
          "sourcePortRanges": [],
          "destinationPortRanges": [],
          "sourceAddressPrefixes": [],
          "destinationAddressPrefixes": []
        }
      },
      ...
    ]
  }
}
```